netlogo-gui/src/test/agent/WorldTests.scala

```diff
@@ -24,6 +26,30 @@ class WorldTests extends FunSuite with AbstractTestWorld {
   override def makeLink(world: World, ends: Array[Int]) =
     new Link(world, world.getTurtle(ends(0)),
              world.getTurtle(ends(1)), world.links)
+
+  def makeBreededTurtle(world: World, breedName: String) =
+    new Turtle(world, world.getBreed(breedName), 0.0, 0.0)
+
+  def makeBreededLink(world: World, breedName: String, end1: Int, end2: Int) =
+    new Link(world, world.getTurtle(end1), world.getTurtle(end2),
+      world.getLinkBreed(breedName))
+
+  def makeWorld(dimensions: WorldDimensions, program: Program) = {
+    val w = new World() {
+      createPatches(dimensions)
+      realloc()
+    }
+    w.program(program)
+    w.realloc()
+    w
+  }
+
+  def changeProgram(w: World, program: Program) = {
+    w.rememberOldProgram()
```

Comment:
I have a few simple tests working on a branch, but I'm going to make a note to go ahead and merge and come back to those tests later for a few reasons:
1. The tests depend on `CompilerManager` being converted to scala
2. I want to get a release out today
I've made myself a note to get those tests merged in and I think they will be quite helpful on the new file format branch, as `CompilerManager` will likely experience some changes on that branch.